src/components/ContinuationIn/ExtraproRegistration.vue

```diff
@@ -237,58 +323,104 @@ export default class ExtraproRegistration extends Vue {
 
-  /** The incorporation date. */
+  /** The incorporation date in the home jurisdiction. */
   get incorporationDate (): string {
-    // FUTURE: format date if needed
-    return this.business.registrationDate
+    // FUTURE: format date as needed
+    return this.business.incorporationDate
   }
 
-  /** Whether business is Active (otherwise it's Historical). */
+  /** Whether the business is Active (otherwise it's Historical). */
   get isBusinessActive (): boolean {
-    return (this.business.status === 'ACTIVE')
+    return (this.business.status === EntityStates.ACTIVE)
   }
 
   mounted (): void {
-    // get the business info object from the store or initialize it
-    this.business = this.getContinuationInBusinessInfo || { ...EmptyBusinessLookup }
+    // set existing business info object from the store, if it exists
+    if (this.getExistingBusinessInfo) this.business = this.getExistingBusinessInfo
 
     // if mode is LOOKUP, set this component to active (which hides the other component)
     if (this.business.mode === 'LOOKUP') this.active = true
   }
 
-  setBusiness (business: BusinessLookupResultIF) {
-    this.business = { ...business } // for reactivity
-    this.business.mode = 'LOOKUP'
-    this.setContinuationInBusinessInfo(this.business)
+  async setBusiness (result: BusinessLookupResultIF): Promise<void> {
+    // FUTURE: fetch extra business data from COLIN and add to business object
+    // NOTE: fetch for both active and historical businesses
+    // const businessInfo = await LegalServices.fetchColinBusinessInfo(business.identifier)
+    //   .catch(() => {})
+
+    // FUTURE: convert FD to FEDERAL and add other COLIN conversions here
+
+    this.business = {
+      homeJurisdiction: null, // get from COLIN
+      businessIdentifier: result.identifier,
+      businessLegalName: result.name,
+      identifier: null, // get from COLIN
+      incorporationDate: null, // get from COLIN
+      legalName: null, // get from COLIN
+      mode: 'LOOKUP',
+      status: result.status,
+      taxId: result.bn?.substring(0, 9) || null
+    }
+    this.setExistingBusinessInfo(this.business)
     this.active = true
   }
 
   /** Resets this component back to its initial state. */
   reset () {
-    this.business = { ...EmptyBusinessLookup }
-    this.setContinuationInBusinessInfo(this.business)
+    this.business = {} as unknown as ExistingBusinessInfoIF
+    this.setExistingBusinessInfo(this.business)
     // set this component to inactive (which shows the other component)
     this.active = false
-    this.checkbox = false
   }
 
-  // *** TODO: add watchers for jurisdiction and affidavit
-  @Watch('business')
-  @Watch('checkbox')
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async fileSelected (file): Promise<void> {
+    // *** TODO: finish implementing -- see UploadMemorandum.vue
+    if (file) {
+      // add property reactively to business object
+      this.$set(this.business, 'affidavitFileKey', '123')
```

Comment:
This is how to add a property reactively to an object (otherwise Vue doesn't react).

On line 394 is how to delete a property reactively.

Reply:
Wait, so `this.business.affidavitFileKey = 123` doesn't work?

Reply:
Your example correctly sets the new property and value, but the watcher on the business object, even with the deep option, doesn't fire.